### PR TITLE
:app + :core — typed InsightSummary + formatter (region-localization infra)

### PR DESCRIPTION
**Pull request by mikelward**
## Summary

Splits the daily insight into a **typed structure** (in `:core:domain`) and a **prose formatter** (in `:app`). This is the **infra half** of a stacked pair; the **feature half** (Region setting + `values-en-rGB/`/`values-en-rAU/` overrides for "sweater" vs "jumper" etc.) lands on top in a separate PR. Cleanly separating the two means this PR is a pure architectural refactor — no user-visible change, prose output is byte-for-byte identical.

## Why split it this way

The user-visible feature is "US users see *sweater*, UK/AU users see *jumper*". To get there cleanly we need:

1. The wardrobe item to flow through as a **key**, not a literal display string.
2. The prose to be assembled where Android string resources are reachable.
3. The cache to store something **re-renderable** so a Region change doesn't need a refetch.

This PR is (1)+(2)+(3). The actual region picker, resource overrides, and key migration land on top.

### `:core:domain`

- **`InsightSummary`** — new data class. One clause per rule, all pure data:
  - `AlertClause(event)`, `BandClause(low, high)`, `DeltaClause(degrees, direction)`, `WardrobeClause(items)`, `PrecipClause(condition, time)`, `CalendarTieInClause(item, time, title)`.
  - `period: ForecastPeriod` carries TODAY vs TONIGHT for the band clause's lead-in.
- **`TemperatureBand`** is now a **public** domain enum (was `internal`) — leaked through `BandClause` so the formatter can resolve it to a localized label.
- **`RenderInsightSummary`** returns `InsightSummary` instead of `String`. Same six-rule logic, same ordering, same edge cases. Tests flip from prose substring to typed assertions.
- **`Insight.summary`** is now `InsightSummary`. The `spokenText()` helper is gone — rendering belongs to `:app`.

### `:app`

- **`InsightFormatter`** — new class, pure Kotlin, produces prose byte-for-byte identical to the previous renderer. Article picker (`a` / `an` / no-article-on-plurals) lives here now. Strings are inline; the feature PR swaps them for `getString` calls.
- **`InsightFormatterTest`** — covers every clause type and the joining rules (Oxford comma, lead-in switch, plural-as-no-article, etc.).
- **`InsightCache`** — schema bumped: `latest_insight_v2` → `v3`, `latest_tonight_insight_v1` → `v2`. Stores the structured `InsightSummary` instead of prose. Pre-v3 prose-summary payloads fail to deserialize against the new schema and drop to null on first read; the next worker run regenerates. **One-time UX cost on update:** users may see a brief "no insight" state if they open the app between update and the next morning alarm.
- **Consumers** — `InsightNotifier`, `TonightInsightNotifier`, `FetchAndNotifyWorker`, `TodayScreen`, `VoiceSettings` (TTS preview) all go through `InsightFormatter` now.
- The `insight.summary.isBlank()` defensive guard in the worker is gone — the structural form always carries a band clause, so blank prose isn't reachable.

### Tests

- `RenderInsightSummaryTest` — restructured (typed assertions per clause).
- `GenerateDailyInsightTest` — same.
- `InsightFormatterTest` — new, in `:app/src/test`. Covers prose assembly.
- `InsightCacheTest` — sample now uses structured summary; new "every clause type round-trips" test; new "pre-v3 payloads drop to null" test.
- `InsightTest` deleted — only tested `spokenText()` which is gone.

## What's *not* in this PR

- No Region enum. No string resources. No `values-en-rGB/`/`values-en-rAU/`. No user-visible change.
- `WardrobeRule.DEFAULTS` still says `"jumper"` — the feature PR flips to `"sweater"` (US baseline) with localized overrides.

I couldn't run `:core:domain:test` locally (sandbox can't reach AGP via Google Maven, per `CLAUDE.md`). Relying on CI as the validation surface.

## Test plan

- [ ] CI: `JVM unit tests` green (new `InsightFormatterTest`, restructured `RenderInsightSummaryTest`, `GenerateDailyInsightTest`, updated `InsightCacheTest`)
- [ ] CI: `Android debug build` green
- [ ] On-device: morning insight at 07:00 fires with prose identical to before (eyeball notification + TTS readout)
- [ ] On-device: tonight insight at 19:00 fires with prose identical to before (silent vs alerting channel still picked correctly by `hasEvents`)
- [ ] On-device: open app after update — "no insight" state for at most one alarm cycle, then regenerates on next worker run
- [ ] On-device: `Settings → Voice → Test voice` plays the spoken summary if a cached insight exists, falls back to the sample string otherwise

---
_Generated by [Claude Code](https://claude.ai/code)_

---
_Generated by [Claude Code](https://claude.ai/code/session_01WuUZhn3A5akttfhWWopzsm)_

### CODE REVIEW

File: app/src/main/kotlin/app/clothescast/insight/InsightFormatter.kt [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+                "$head, $middle, and $tail"
+            }
+        }
+        return "Wear $phrase."
+    }
+
+    private fun formatPrecip(precip: PrecipClause): String {
+        val type = precip.condition.name.lowercase(Locale.ROOT).replace('_', ' ')
+            .replaceFirstChar { it.titlecase(Locale.ROOT) }
+        return "$type at ${EVENT_TIME.format(precip.time)}."
```
[the remark this answers is not in the document]
Comment (mikelward) [reply app/src/main/kotlin/app/clothescast/insight/InsightFormatter.kt:73]: Good catch — fixed in `518220f`. The previous renderer used `LocalTime.toString()` for precip (always ISO 8601 ASCII) and only routed calendar tie-in through the formatter, but the new formatter shares one `EVENT_TIME` instance for both. Pinning it to `Locale.ROOT` keeps digits ASCII regardless of device locale and is a strict improvement over the previous calendar-tie-in behaviour, which was already locale-default and would have shaped digits under e.g. Arabic locales.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: app/src/test/kotlin/app/clothescast/insight/InsightFormatterTest.kt [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    }
+
+    @Test
+    fun `delta clause emits cooler`() {
+        val out = subject.format(summary(delta = DeltaClause(6, DeltaClause.Direction.COOLER)))
+        out shouldBe "Today will be mild. It will be 6° cooler today."
+    }
+
+    @Test
+    fun `wardrobe with a single article-able item emits 'a item'`() {
```
[the remark this answers is not in the document]
Comment (mikelward) [reply app/src/test/kotlin/app/clothescast/insight/InsightFormatterTest.kt]: Renamed in `518220f` — the test is now `wardrobe with a single article-able item emits 'a <item>'`.

---
_Generated by [Claude Code](https://claude.ai/code)_

[the remark this answers is not in the document]
Comment (mikelward) [reply app/src/test/kotlin/app/clothescast/insight/InsightFormatterTest.kt]: Quick follow-up on this — the rename to `'a <item>'` failed CI (`Name contains illegal characters: <>`; the JVM rejects angle brackets in method names even when Kotlin source uses backticks). Fixed in `3b37b21` by switching to the other half of your suggestion, `'a jumper'`, which lined up with the assertion in the test body anyway.

---
_Generated by [Claude Code](https://claude.ai/code)_
